src/utils/automation-helper.js

```diff
@@ -29,10 +29,337 @@ export const getAutomation = async (automationId) => {
+  if (Array.isArray(data)) {
+    return { message: data.join(' ').length > 0 ? data.join(' ') : defaultMessage };
+  }
+  if (Object.keys(data).includes('message')) {
+    return data;
+  }
+  return { message: defaultMessage };
+};
+
+const parseOutput = (data, translator) => {
+  if (typeof data !== 'undefined') {
+    try {
+      const message = JSON.parse(data);
+      return checkMessage(message, translator);
+    } catch {
+      if (typeof data === 'object') {
+        return checkMessage(data, translator);
+      }
+      return { message: data };
+    }
+  }
+  return { message: translator.get('Object.ActionButton.Automation.DefaultAutomationMsg') };
+};
+
+// eslint-disable-next-line no-promise-executor-return
+const sleep = (time) => new Promise((resolve) => setTimeout(resolve, time));
+
+const automationRunPolling = async (automationId, runId, translator) => {
+  const runningStatuses = ['queued', 'running', 'not started', 'starting'];
+  // Max sleep time ~10 minutes
+  const sleepTimes = [
+    2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 15, 15, 15, 15, 15, 15,
+    30, 30, 30, 30, 60, 60, 60, 60, 60, 60,
+  ];
+  let finalRun;
+  for (let i = 0; i < sleepTimes.length; i++) {
+    // eslint-disable-next-line no-await-in-loop
+    const automationRun = await getAutomationRun(automationId, runId);
+    const { status } = automationRun;
+    if (!runningStatuses.includes(status)) {
+      finalRun = automationRun;
+      break;
+    } else {
+      // eslint-disable-next-line no-await-in-loop
+      await sleep(sleepTimes[i] * 1000);
+    }
+  }
+  let msg;
+  switch (finalRun.status) {
+    case 'finished': {
+      if (finalRun.title?.length > 0) {
+        msg = parseOutput(finalRun.title, translator);
+        msg.ok = true;
+      } else {
+        msg = { message: translator.get('Object.ActionButton.Automation.DefaultAutomationMsg'), ok: true };
+      }
+      break;
+    }
+    case 'failed': {
+      if (finalRun.title?.length > 0) {
+        msg = parseOutput(finalRun.title, translator);
+        msg.ok = false;
+      } else {
+        msg = { message: translator.get('Object.ActionButton.Automation.AutomationError'), ok: true };
+      }
+      break;
+    }
+    case 'finished with warnings': {
+      if (finalRun.title?.length > 0) {
+        msg = parseOutput(finalRun.title, translator);
+        msg.ok = false;
+      } else {
+        msg = { message: translator.get('Object.ActionButton.Automation.DefaultAutomationMsg'), ok: true };
+      }
+      break;
+    }
+    case 'must stop':
+    case 'stopped': {
+      if (finalRun.title?.length > 0) {
+        msg = parseOutput(finalRun.title, translator);
+        msg.ok = false;
+      } else {
+        msg = { message: translator.get('Object.ActionButton.Automation.UnkownError'), ok: true };
+      }
+      break;
+    }
+    default: {
```

Comment:
This default also gets triggered when you run out of the for loop. You will get a message saying automation finished when it actually is not finished yet I think

Reply:
In the default case the automation has finished running but does not have a status which are in the cases defined - We should never actually run into the default case but if we do "Automation finished" is not totally incorrect so I think this is fine